rpcs3/Emu/System.cpp

```diff
@@ -505,17 +505,18 @@ namespace
 
 bool Emulator::SetUsr(const std::string& user)
 {
-	if (user.empty())
-	{
-		return false;
-	}
+	sys_log.notice("Setting user ID '%s'", user);
 
 	u32 id = 0;
-	std::from_chars(&user.front(), &user.back() + 1, id);
+
+	if (user.size() == 8)
+	{
+		std::from_chars(&user.front(), &user.back() + 1, id);
+	}
 
 	if (id == 0)
 	{
-		return false;
+		fmt::throw_exception("Failed to set user ID '%s'.\nIt must consist of 8 digits and cannot be 00000000.", user);
```

Comment:
throw_exception in main thread crashes the whole emulator... just log an error and return false. As a general practice use assertations only if there's absolutely no logical way the thread could continue normally after the unexpected condition has been met, or if it's really an unreachable condition by your perspective.

Reply:
updated.
There shouldn't be a code path that isn't properly controlled now.